### DISCUSSION TITLE
Import Request untuk Menarik Data dari Link API (Output: Response [401] dan [404])

### DISCUSSION THREAD

**Discussion by ki-arief** [teamalgoritma/community#823] [Q&A]
Saya mencoba menarik data komoditas energi dari API (berlangganan).

![Capture1](https://github.com/teamalgoritma/community/assets/150775927/5e9477a7-5094-42c2-b598-595e4747331b)
kemudian muncul output (Response [401])

sudah diakukan autentifikasi di website link API tersebut.
![Capture2](https://github.com/teamalgoritma/community/assets/150775927/f332deb3-925e-480b-a120-3492ff724253)
kemudian muncul output (Response [404])

Bagaimana cara untuk menampilkan data tersebut?

**Comment by ramdhinafinita** [comment]
Dear @ki-arief 

Berdasarkan error yang muncul karena API yang digunakan berlangganan kami tidak bisa akses API tersebut walau mencari link platts itu sendiri. 
Alasan 401:
1. Missing API -> platts API membutuhkan API key untuk mengakses data tersebut. Anda perlu melakukan registrasi terlebih dahulu 
2. Jika sudah, dan melakukan authentifikasi -> cek apakah API KEY nya sudah dimasukkan kedalam headernya. 

```
headers = {
    'accept': 'application/json',
    'Content-Type': 'application/x-www-form-urlencoded',
    'Authorization': 'Bearer YOUR_API_KEY',  # Replace with your actual API key
}
```

**Reply by ki-arief** [reply]
Sudah dijalankan cek API Key ini. 
![Capture3](https://github.com/teamalgoritma/community/assets/150775927/dc5a162b-c37e-4cce-a4e3-761cc0b16f60)

cuma untuk request API nya masih [401]

**Comment by finesaaa** [comment]
Disini seharusnya tutorial dari Platts nya cukup lengkap pak, boleh diikuti step by stepnya https://view.highspot.com/viewer/640603b2f7440cd78a64f293
